### PR TITLE
check Pipeline API version compatibility before sending any requests

### PR DESCRIPTION
**Pull request by siliconbrain**
.
### ToDo
* [ ] make API compatible with generated client

### CODE REVIEW

File: internal/cli/cli.go [hunk 1]
```diff
@@ -132,6 +135,15 @@ func (c *banzaiCli) Client() *pipeline.APIClient {
 		config.HTTPClient.Transport.(*oauth2.Transport).Base = c.RoundTripper()
 
 		c.client = pipeline.NewAPIClient(config)
+
+		resp, _, err := c.client.CommonApi.ApiVersionGet(context.Background())
+		if err != nil {
+			LogAPIError("query API version", err, nil)
+			log.Fatalf("failed to query API version: %v", err)
+		}
+		if ver := resp.Version; ver != supportedPipelineAPIVersion {
```
Comment (orymate) [inline internal/cli/cli.go]: exact match is a no-go for me

Comment (sagikazarmark) [reply internal/cli/cli.go]: https://pkg.go.dev/github.com/Masterminds/semver/v3?tab=doc#Version.Compare

Comment (siliconbrain) [reply internal/cli/cli.go]: Yeah, I know that we should check semver, but what would the semantics be?

Comment (orymate) [reply internal/cli/cli.go]: we know the first version that is compatible with this cli code. we should check if the server version is equal or greater than it.

Comment (siliconbrain) [reply internal/cli/cli.go]: But a newer server version might not be compatible with this CLI code.

**Pull request closed without merging**